tests/integ/test_stored_procedure.py

```diff
@@ -689,6 +695,145 @@ def _(
     assert "Pandas stored procedure is not supported" in str(ex_info)
 
 
+@pytest.mark.parametrize("is_permanent", [True, False])
+@pytest.mark.parametrize("anonymous", [True, False])
+@pytest.mark.parametrize(
+    "ret_type",
+    [
+        StructType(),
+        StructType(
+            [
+                StructField("a", StringType()),
+                StructField("b", StringType()),
+                StructField("c", StringType()),
+            ]
+        ),
+    ],
+)
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="Named temporary procedure is not supported in stored proc",
+)
+def test_table_sproc(session, is_permanent, anonymous, ret_type):
+    """Ensure the following scenarios work:
+    - register sproc with session.sproc.register
+    - register sproc with @sproc decorator
+    - can create permanent and temporary sprocs
+    - can create anonymous sprocs
+    - session.call works with provided function name
+    - we can both specify return cols types and keep it blank
+        - exception: sproc from decorator and implicit type hint cannot specify return col types
+    - dataframe returned after a sproc call can be operated on like normal dataframes
+    """
+    if len(ret_type.fields) == 0 and not session.sql_simplifier_enabled:
+        # if return type does not define output columns and sql_simplifier is
+        # disabled, then we don't support dataframe operations on table sprocs
+        pytest.skip()
+
```

Comment:
I want to know how the team feels about this. The reason why we cannot have dataframe operations working with sql simplifier disabled + return columns not defined is because when sql simplifier is disabled, the code path triggers `plan.output` which results in describe query for the plan `call table_sproc(args)`. Describe query only runs describe without running the actual functions with arguments. In the case when output columns are not defined in table sproc functions definition, the describe query returns `TABLE ()`. This breaks subsequent query creation. 
The same does not happen with sql simplifier is enabled because it skips the part which calls `plan.output`.